File: codegen/results/v001/gemma4_26b_think/20260407_165540/go/004_json_filter/solution.go
package main

import (
	"encoding/json"
	"fmt"
	"os"
	"sort"
)

// Record represents the structure of the objects in the input JSON array.
type Record struct {
	Name   string  `json:"name"`
	Age    int     `json:"age"`
	Active bool    `json:"active"`
	Score  float64 `json:"score"`
}

func main() {
	// 1. Read the file input/data.json
	content, err := os.ReadFile("input/data.json")
	if err != nil {
		return
	}

	// 2. Parse the JSON array
	var records []Record
	err = json.Unmarshal(content, &records)
	if err != nil {
		return
	}

	// 3. Filter records: active is true and age is 30 or older
	var filtered []Record
	for _, r := range records {
		if r.Active && r.Age >= 30 {
			filtered = append(filtered, r)
		}
	}

	// 4. Sort the filtered records by name ascending
	sort.Slice(filtered, func(i, j int) bool {
		return filtered[i].Name < filtered[j].Name
	})

	// 5. Output the resulting JSON array to stdout
	output, err := json.MarshalIndent(filtered, "", "  ")
	if err != nil {
		return
	}

	fmt.Print(string(output))
}

File: codegen/results/v001/gemma4_26b_think/20260407_165540/typescript/004_json_filter/solution.ts
import * as fs from 'fs';
import * as path from 'path';

/**
 * Interface representing the structure of the input records.
 */
interface Record {
  name: string;
  age: number;
  active: boolean;
  score: number;
}

/**
 * Main function to process the JSON data.
 */
function processData(): void {
  // Define the path to the input file as specified.
  const inputPath = path.resolve(process.cwd(), 'input', 'data.json');

  try {
    // Check if the file exists before attempting to read.
    if (!fs.existsSync(inputPath)) {
      return;
    }

    // Read the file content synchronously.
    const fileContent = fs.readFileSync(inputPath, 'utf8');
    
    // Parse the JSON array.
    const data: Record[] = JSON.parse(fileContent);

    // Apply filters: 
    // 1. active must be true.
    // 2. age must be 30 or older.
    const filteredData = data.filter((item) => item.active === true && item.age >= 30);

    // Sort the resulting array by name in ascending order.
    filteredData.sort((a, b) => a.name.localeCompare(b.name));

    // Output the final JSON array to stdout.
    // Using JSON.stringify without indentation to ensure no extra whitespace or newlines are added.
    process.stdout.write(JSON.stringify(filteredData) + '\n');
  } catch (error) {
    // In case of a parsing error or file access error, we exit without printing to stdout
    // to comply with the "Do not output anything else" requirement.
    process.exit(1);
  }
}

// Execute the processing function.
processData();